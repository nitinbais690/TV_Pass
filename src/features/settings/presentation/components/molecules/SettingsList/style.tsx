import { StyleSheet } from 'react-native';
import { scale, selectDeviceType } from 'qp-common-ui';

export function settingsListStyle() {
    return StyleSheet.create({
        iconStyle: {
            width: selectDeviceType({ Handset: scale(24, 0) }, scale(34, 0)),
            height: selectDeviceType({ Handset: scale(24, 0) }, scale(34, 0)),
        },
    });
}
