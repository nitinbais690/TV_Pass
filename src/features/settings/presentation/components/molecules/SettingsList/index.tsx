import React from 'react';
import { View } from 'react-native';
import { NAVIGATION_TYPE } from 'screens/Navigation/NavigationConstants';
import SettingsItemRow from '../../atoms/SettingsItemRow';
import { settingsListStyle } from './style';
import Payment from 'assets/images/payment_outline.svg';
import Device from 'assets/images/vibration_outline.svg';
import Account from 'assets/images/account_outline.svg';
import Padlock from 'assets/images/padlock_outline.svg';
import Gear from 'assets/images/settings.svg';
import { useLocalization } from 'contexts/LocalizationContext';

const SettingsList = (props: SettingsListProps) => {
    const { strings } = useLocalization();
    const styles = settingsListStyle();

    const rowItems: Array<RowItem> = [
        {
            title: `${strings['settings.list.subscription_details.title']}`,
            subtitle: `${strings['settings.list.subscription_details.subtitle']}`,
            screen: NAVIGATION_TYPE.SUBSCRIPTION_DETAILS,
        },
        {
            title: `${strings['settings.list.device_management.title']}`,
            subtitle: `${strings['settings.list.device_management.subtitle']}`,
            screen: NAVIGATION_TYPE.MANAGE_DEVICES,
        },
        {
            title: `${strings['settings.list.user_profiles.title']}`,
            subtitle: `${strings['settings.list.user_profiles.subtitle']}`,
            screen: NAVIGATION_TYPE.USER_PROFILE,
        },
        {
            title: `${strings['settings.list.parental_controls.title']}`,
            subtitle: `${strings['settings.list.parental_controls.subtitle']}`,
            screen: NAVIGATION_TYPE.PARENTAL_CONTROLS,
        },
        {
            title: `${strings['settings.list.app_settings.title']}`,
            subtitle: `${strings['settings.list.app_settings.subtitle']}`,
            screen: NAVIGATION_TYPE.APP_SETTINGS,
        },
    ];

    function getRowIcon(row: string) {
        switch (row) {
            case NAVIGATION_TYPE.SUBSCRIPTION_DETAILS:
                return <Payment height={styles.iconStyle.height} width={styles.iconStyle.width} />;
            case NAVIGATION_TYPE.MANAGE_DEVICES:
                return <Device height={styles.iconStyle.height} width={styles.iconStyle.width} />;
            case NAVIGATION_TYPE.USER_PROFILE:
                return <Account height={styles.iconStyle.height} width={styles.iconStyle.width} />;
            case NAVIGATION_TYPE.PARENTAL_CONTROLS:
                return <Padlock height={styles.iconStyle.height} width={styles.iconStyle.width} />;
            case NAVIGATION_TYPE.APP_SETTINGS:
                return <Gear height={styles.iconStyle.height} width={styles.iconStyle.width} />;
        }
    }

    const renderItem = (item: RowItem) => {
        return (
            <SettingsItemRow
                key={item.screen}
                title={item.title}
                subtitle={item.subtitle}
                image={getRowIcon(item.screen)}
                onPress={() => {
                    props.onPress(item);
                }}
            />
        );
    };

    return <View>{rowItems.map(item => renderItem(item))}</View>;
};

export interface RowItem {
    title: string;
    subtitle: string;
    [key: string]: any;
}

interface SettingsListProps {
    onPress: (screen: RowItem) => void;
}

export default SettingsList;
