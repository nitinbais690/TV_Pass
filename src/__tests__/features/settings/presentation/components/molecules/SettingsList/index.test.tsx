import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import SettingsList from 'features/settings/presentation/components/molecules/SettingsList';

jest.mock('core/presentation/hooks/use-app-colors');

describe('settingsList', () => {
    it('renders correctly', () => {
        const pressHandler = jest.fn();
        const { container } = render(<SettingsList onPress={pressHandler} />);
        expect(toJSON(container)).toMatchSnapshot();
    });
});
