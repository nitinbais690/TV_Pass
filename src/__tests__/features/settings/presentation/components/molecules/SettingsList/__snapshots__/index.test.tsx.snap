// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settingsList renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      hasTVPreferredFocus={false}
      isTVSelectable={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#FFFFFF",
              "opacity": 0.07,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingTop": 32,
            }
          }
        >
          <SvgMock
            height={34}
            width={34}
          />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "ProximaNova-Semibold",
                "fontSize": 20,
                "fontWeight": "600",
                "paddingLeft": 20,
              }
            }
          >
            Subscription Details
          </Text>
          <SvgMock
            style={
              Object {
                "height": 34,
                "marginStart": "auto",
                "width": 34,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "ProximaNova-Semibold",
              "fontSize": 18,
              "fontWeight": "600",
              "opacity": 0.3,
              "paddingBottom": 26,
              "paddingTop": 9,
            }
          }
        >
          Manage your subscription & payment
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      hasTVPreferredFocus={false}
      isTVSelectable={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#FFFFFF",
              "opacity": 0.07,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingTop": 32,
            }
          }
        >
          <SvgMock
            height={34}
            width={34}
          />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "ProximaNova-Semibold",
                "fontSize": 20,
                "fontWeight": "600",
                "paddingLeft": 20,
              }
            }
          >
            Device Management
          </Text>
          <SvgMock
            style={
              Object {
                "height": 34,
                "marginStart": "auto",
                "width": 34,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "ProximaNova-Semibold",
              "fontSize": 18,
              "fontWeight": "600",
              "opacity": 0.3,
              "paddingBottom": 26,
              "paddingTop": 9,
            }
          }
        >
          Manage all your devices
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      hasTVPreferredFocus={false}
      isTVSelectable={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#FFFFFF",
              "opacity": 0.07,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingTop": 32,
            }
          }
        >
          <SvgMock
            height={34}
            width={34}
          />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "ProximaNova-Semibold",
                "fontSize": 20,
                "fontWeight": "600",
                "paddingLeft": 20,
              }
            }
          >
            User Profiles
          </Text>
          <SvgMock
            style={
              Object {
                "height": 34,
                "marginStart": "auto",
                "width": 34,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "ProximaNova-Semibold",
              "fontSize": 18,
              "fontWeight": "600",
              "opacity": 0.3,
              "paddingBottom": 26,
              "paddingTop": 9,
            }
          }
        >
          Manage your user profiles here
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      hasTVPreferredFocus={false}
      isTVSelectable={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#FFFFFF",
              "opacity": 0.07,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingTop": 32,
            }
          }
        >
          <SvgMock
            height={34}
            width={34}
          />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "ProximaNova-Semibold",
                "fontSize": 20,
                "fontWeight": "600",
                "paddingLeft": 20,
              }
            }
          >
            Parental Controls
          </Text>
          <SvgMock
            style={
              Object {
                "height": 34,
                "marginStart": "auto",
                "width": 34,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "ProximaNova-Semibold",
              "fontSize": 18,
              "fontWeight": "600",
              "opacity": 0.3,
              "paddingBottom": 26,
              "paddingTop": 9,
            }
          }
        >
          Manage viewing restrictions
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      hasTVPreferredFocus={false}
      isTVSelectable={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#FFFFFF",
              "opacity": 0.07,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingTop": 32,
            }
          }
        >
          <SvgMock
            height={34}
            width={34}
          />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "ProximaNova-Semibold",
                "fontSize": 20,
                "fontWeight": "600",
                "paddingLeft": 20,
              }
            }
          >
            App Settings
          </Text>
          <SvgMock
            style={
              Object {
                "height": 34,
                "marginStart": "auto",
                "width": 34,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "ProximaNova-Semibold",
              "fontSize": 18,
              "fontWeight": "600",
              "opacity": 0.3,
              "paddingBottom": 26,
              "paddingTop": 9,
            }
          }
        >
          Manage your app from here
        </Text>
      </View>
    </View>
  </View>
</View>
`;
